xml/System.IO.Pipelines/PipeReader.xml

```diff
@@ -264,10 +264,19 @@
         <Parameter Name="state" Type="System.Object" />
       </Parameters>
       <Docs>
-        <param name="callback">To be added.</param>
-        <param name="state">To be added.</param>
-        <summary>Cancels the pending <see cref="M:System.IO.Pipelines.PipeReader.ReadAsync(System.Threading.CancellationToken)" /> operation. If there is none, cancels next <see cref="M:System.IO.Pipelines.PipeReader.ReadAsync(System.Threading.CancellationToken)" /> operation, without completing the <see cref="T:System.IO.Pipelines.PipeWriter" />.</summary>
-        <remarks>To be added.</remarks>
+        <param name="callback">The callback to register.</param>
+        <param name="state">The state object to pass to <paramref name="callback" /> when it's invoked.</param>
+        <summary>Registers a callback that gets executed when the <see cref="T:System.IO.Pipelines.PipeWriter" /> side of the pipe is completed.</summary>
+         <remarks>
+           <format type="text/markdown"><![CDATA[
+
+## Remarks
+
+> [!IMPORTANT]
+> `OnWriterCompleted` may not be invoked on all implementations of <xref:System.IO.Pipelines.PipeWriter>. This will be removed in a future release.
```

Comment:
```suggestion
> `OnWriterCompleted` may not be invoked on all implementations of <xref:System.IO.Pipelines.PipeWriter>. This limitation will be removed in a future release.
```
Here and below, can we specify what implementations don't support it? If it throws PlatformNotSupportedException or NotImplementedException, that should be noted in exceptions.

Reply:
@rpetrusha:

Here is [the source code](https://github.com/dotnet/corefx/blob/51cd78434953683040f93e69ed2a889e476f5cac/src/System.IO.Pipelines/src/System/IO/Pipelines/PipeWriter.cs) for this function.

I looked at the git blame for the "Obsolete" attribute line, and it was added [11 days ago in this PR](https://github.com/dotnet/corefx/pull/39526/files). That change changed both the `OnWriterCompleted` and `OnReaderCompleted` methods from abstract to virtual, added an empty body, and added the Obsolete attribute on top of both methods.

But the actual function was added 2 years ago [in this commit](https://github.com/dotnet/corefx/commit/5573f26fc34ba040b03f022a5892a5e172573002). So the function is now new, but I suppose that since it was abstract, and now it's virtual, is the reason why it showed up in the Docs spreadsheet with the dotnet-plat-ext-3.0 moniker (but I'm not entirely sure this is the actual reason).

Based on this, would this message make more sense?:

```suggestion
> `OnWriterCompleted` may not be invoked on .NET Core 3.0 implementations of <xref:System.IO.Pipelines.PipeWriter>. This limitation will be removed in a future release.
```

@davidfowl can you chime in, since you made the change to this API? Am I understanding this correctly?

Reply:
No that's not correct. We actually want to stop people from using this API because it cannot be implemented in some cases.

Reply:
Understood. @davidfowl so would you prefer us to keep the original sentence? Would you like us to reword it a bit?

Reply:
Feel free to word smith it a bit. That was the best i could come up with 😄

Reply:
Okay, so we should not use `limitation` as it was originally suggested. If the intention is to stop people using the API, I think this would clarify the last sentence:

```suggestion
> `OnWriterCompleted` may not be invoked on all implementations of <xref:System.IO.Pipelines.PipeWriter>. This method will be removed in a future release.
```

Reply:
Yep

Reply:
I think that this could be made stronger, @carlossanlop and @davidfowl. First, it could go in an IMPORTANT note box. And it could read, "We recommend that you not call this method, since it may not be invoked on all implementations of \<xref:System.IO.PipeLines.PipeWriter>. The method will be removed in a future release of .NET Core.

Reply:
@rpetrusha I like your suggestion, it's stronger as recommended, while preserving the original idea @davidfowl intended in the comment. If David's ok with the change, we should change it:

```suggestion
We recommend that you not call this method, since it may not be invoked on all implementations of <xref:System.IO.PipeLines.PipeWriter>. The method will be removed in a future release of .NET Core.
```